lib/Twig/Extension/Templating/PimcoreUrl.php

```diff
@@ -90,13 +89,14 @@ protected function generateUrl(array|string $name = null, ?array $parameters = [
             $name = $this->getCurrentRoute();
         }
 
+        /** @var Concrete $object */
         $object = $parameters['object'] ?? null;
         $linkGenerator = null;
 
-        if ($object instanceof LinkGeneratorAwareInterface) { //e.g. Mockup
-            $linkGenerator = $object->getLinkGenerator();
-        } elseif ($object instanceof Concrete) {
+        if (method_exists($object, 'getClass')) {
             $linkGenerator = $object->getClass()->getLinkGenerator();
```

Comment:
We should also check if `getLinkGenerator()` method exists. Since object can be anything.